oximeter/db/src/db-replicated-init.sql

```diff
@@ -18,9 +18,7 @@ CREATE TABLE IF NOT EXISTS oximeter.measurements_bool ON CLUSTER oximeter_cluste
     timestamp DateTime64(9, 'UTC'),
     datum UInt8
 )
-ENGINE = Distributed('oximeter_cluster', 'oximeter', 'measurements_bool_local', timeseries_name)
-ORDER BY (timeseries_name, timeseries_key, timestamp)
-TTL toDateTime(timestamp) + INTERVAL 30 DAY;
+ENGINE = Distributed('oximeter_cluster', 'oximeter', 'measurements_bool_local', timeseries_key);
```

Comment:
Yeah, that was intentional. Unfortunately, there was a `DB::Exception` error because the sharding key cannot be a string, it must be a uint. Using timeseries_key made sense to me, but can change it to something else if you think that would make more sense.

Reply:
Hmm, bummer that the key needs to be an integer. In that case, I'd recommend we shard by the hash of the _target name_. This is all pretty abstract until we actually shard the data, but here's the rationale.

We'd like to group data together that is likely to be related. I'd expect the most relevant grouping to be targets. For example, if we're measuring a bunch of stuff about a disk drive, it seems more likely that we want all the information about the drive than some other random target. That is, I'd expect we're more likely to run queries like, "show me the temperature and write latency for this disk drive" compared to queries like "show me the temperature for this disk drive and the voltage on this sensor." To be clear, correlations across targets are going to be critical too. And we can always answer them, regardless of sharding key. I just expect the former kind of query to be more likely at this point.

Assuming that's true, then something like `xxHash64(splitByChar(':', timeseries_name)[1])` seems reasonable. That should distribute all metrics for the same targets to the same shard. Different targets will be randomly distributed.

Reply:
I'm not entirely convinced we're more likely to run queries on single targets than on correlated targets. Back when I was an SRE I remember correlating several metrics in order to get a clear picture of what was going on. 

That said, I don't see how using the hashed timeseries_name as sharding key could have a negative impact on those queries. Happy to change the sharding key to the hashed name :)

Reply:
Here's another way to argue for hashing the target. If you hash the full timeseries name, we're only guaranteed that timeseries with the exact same name are on the same node. So you don't get a locality benefit from either the target name or metric name. If we hash the target name, we still don't get any locality in the metrics or full timeseries, but at least we get it for the targets. So we get _some_ locality, whereas hashing the full name gives us strictly less locality.